sklearn/utils/tests/test_extmath.py

```diff
@@ -161,6 +162,94 @@ def test_randomized_svd_low_rank_all_dtypes(dtype):
     check_randomized_svd_low_rank(dtype)
 
 
+@pytest.mark.parametrize('dtype',
+                         (np.int32, np.int64, np.float32, np.float64))
+def test_randomized_eigsh(dtype):
+    """Test that `_randomized_eigsh` returns the appropriate components"""
+
+    rng = np.random.RandomState(42)
+    X = np.diag(np.array([1., -2., 0., 3.], dtype=dtype))
+    # random rotation that preserves the eigenvalues of X
+    rand_rot = np.linalg.qr(rng.normal(size=X.shape))[0]
+    X = rand_rot @ X @ rand_rot.T
+
+    # with 'module' selection method, the negative eigenvalue shows up
+    lambd_, alph_ = _randomized_eigsh(X, n_components=2, selection='module')
```

Comment:
Here again, please use `eigvals` and `eigvecs` instead of the obscure `lambd_` and `alph_` variable names.

Ok for keeping alpha and lambda in the `KernelPCA` class for the sake of backward compat, but let's not propagate a bad choice of variable names to other places in the code base :)